app/src/main/java/io/gnosis/safe/ui/safe/terms/TermsViewModel.kt

```diff
@@ -0,0 +1,31 @@
+package io.gnosis.safe.ui.safe.terms
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import javax.inject.Inject
+
+class TermsViewModel @Inject constructor(
+    private val termsChecker: TermsChecker
+) : ViewModel() {
+
+    lateinit var advance: () -> Unit
```

Comment:
I seems to me you don't need this `var`. You update it, if `termsChecker.getTermsAgreed` returns false, but when it returns true, you call the method passed in the argument. `invoke()` gets never called for this reference, as far as I can see.

Reply:
It gets called in two cases:
1. Terms have already been agreed and that has been stored in preferences
2. When the user clicks on the agree button

Reply:
advance is removed in 09ed4521d64e0e7151f5e6bdeb4794b9d1622861